uaclient/jobs/gcp_auto_attach.py

```diff
@@ -2,15 +2,25 @@
 Try to auto-attach in a GCP instance. This should only work
 if the instance has a new UA license attached to it
 """
+import base64
+import json
 import logging
 
 from uaclient import config, exceptions
 from uaclient.cli import action_auto_attach
+from uaclient.clouds.gcp import GCP_LICENSES, UAAutoAttachGCPInstance
 from uaclient.clouds.identity import get_cloud_type
+from uaclient.util import get_platform_info
 
 LOG = logging.getLogger(__name__)
 
 
+def get_licenses_from_identity(identityToken):
```

Comment:
This function feels tightly coupled with other GCP metadata specific handling. It's probably best to keep this in uaclient.clouds.gcp module so "future us" can see the impact locally if we end up altering gcp.identity_doc function to return different outputs/format/key names etc.